test/libsolidity/syntaxTests/errors/error_selector_syntax.sol

```diff
@@ -1,48 +1,34 @@
 library L {
-    error E();
-}
-library S {
-    error E(uint);
-}
-library T {
-    error E();
+    error E(bytes4, bool, bytes);
 }
 
-error E();
+error E(bytes4, bool, bytes);
 
 interface I {
-    error E();
-    function f() external pure;
+    error E(bytes4, bool, bytes);
 }
 
-contract D {
-    error F();
+contract B {
+    error E(bytes4, bool, bytes);
 }
 
-contract C is D {
-    function test1() public pure returns (bytes4, bytes4, bytes4, bytes4) {
-        assert(L.E.selector == T.E.selector);
-        assert(L.E.selector != S.E.selector);
-        assert(E.selector == L.E.selector);
-        assert(I.E.selector == L.E.selector);
-        return (L.E.selector, S.E.selector, E.selector, I.E.selector);
-    }
+contract C is B {
+    bytes4 public librarySelector = L.E.selector;
+    bytes4 internal freeSelector = E.selector;
+    bytes4 internal contractSelector = B.E.selector;
+    bytes4 private interfaceSelector = I.E.selector;
 
-    bytes4 s1 = L.E.selector;
-    bytes4 s2 = S.E.selector;
-    bytes4 s3 = T.E.selector;
-    bytes4 s4 = I.E.selector;
-
-    function test2() view external returns (bytes4, bytes4, bytes4, bytes4) {
-        return (s1, s2, s3, s4);
-    }
+    function f(bool condition) public view {
+        assert(librarySelector == L.E.selector);
+        assert(E.selector == B.E.selector);
 
-    function test3() pure external returns (bytes4) {
-        return (F.selector);
+        if (condition)
+            revert E(E.selector, true, "123");
+        else
+            revert L.E((B.E.selector), true, "123");
     }
```

Comment:
I have rewritten this test for errors and the other, almost identical one for events because they we already have almost identical semantic tests. They weren't really testing all that much about the selector syntax. I made it a bit more diverse.